Add form cache

kzdnk opened the pull request:
Adds saving input values to localstorage and restoring them from localstorage on page reload

kzdnk committed:
Add form cache

krychaxp commented on src/components/Form/Form.tsx:
```diff
@@ -46,18 +47,48 @@ const schema = z.object({
     mode: 'onBlur',
   });
 
+  useEffect(() => {
```
Use external library (for example, react-use useLocalStorage)

krychaxp commented on src/components/Form/Form.tsx:
```diff
@@ -46,18 +47,48 @@ const schema = z.object({
+      const defaultValues = localStorage.getItem('defaultValues');
+      if (defaultValues) {
+        for (const [name, value] of Object.entries(JSON.parse(defaultValues))) {
+          setValue(name as InputNames, value as string);
```
Don't use "as string"

Safeheaded commented on src/components/Form/Form.tsx:
```diff
@@ -46,18 +47,48 @@ const schema = z.object({
 
 export type SchemaType = z.TypeOf<typeof schema>;
 
+type InputNames =
```
Zamiast oddzielnie definiować całkowicie nowy typ, lepiej zrobić `type InputNames = keyof SchemaType;`. Wtedy automatycznie wszystkie nazwy pół wyciągnie z już istniejącego interfejsu i nie trzeba będzie zmieniać dwa razy - mniejsza szansa na bug'a w przyszłości

Safeheaded reviewed (changes requested):
Dobrze by było też od razu dodać jakiś przycisk, który resetuje localStorage. Nie musi być na razie łatwy, byleby tylko była taka opcja

kzdnk committed:
Use narrowing instead of 'as string'

kzdnk committed:
Refactor cache to use useLocalStorage

kzdnk committed:
Rename 'defaultValues' to 'savedValues'

kzdnk committed:
Fix: Maximum update depth exceeded

This effect should run only once on startup so theres not need for it to
depend on 'savedValues'.

kzdnk committed:
Add button for reseting the form
